pkg/bosh/manifest/dns.go

```diff
@@ -0,0 +1,323 @@
+	}
+	service := v1.Service{
+		ObjectMeta: metadata,
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{Name: "dns", Port: 53, Protocol: "UDP", TargetPort: intstr.FromInt(8053)},
+				{Name: "dns-tcp", Port: 53, Protocol: "TCP", TargetPort: intstr.FromInt(8053)},
+				{Name: "metrics", Port: 9153, Protocol: "TCP", TargetPort: intstr.FromInt(9153)},
+			},
+			Selector: map[string]string{"app": "bosh-dns"},
+			Type:     "ClusterIP",
+		},
+	}
+
+	var mode int32 = 420
+	deployment := v13.Deployment{
+		ObjectMeta: metadata,
+		Spec: v13.DeploymentSpec{
+			Selector: &v12.LabelSelector{
+				MatchLabels: map[string]string{"app": "bosh-dns"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: v12.ObjectMeta{
+					Labels: map[string]string{"app": "bosh-dns"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "coredns",
+							Args:  []string{"-conf", "/etc/coredns/Corefile"},
+							Image: "coredns/coredns:1.6.3",
+							Ports: []v1.ContainerPort{
+								{ContainerPort: 8053, Name: "dns-udp", Protocol: "UDP"},
+								{ContainerPort: 8053, Name: "dns-tcp", Protocol: "TCP"},
+								{ContainerPort: 9153, Name: "metrics", Protocol: "TCP"},
+							},
+							VolumeMounts: []v1.VolumeMount{
+								{MountPath: "/etc/coredns", Name: "bosh-dns-volume", ReadOnly: true},
+							},
+						},
+					},
+					Volumes: []v1.Volume{
+						{
+							Name: "bosh-dns-volume",
+							VolumeSource: v1.VolumeSource{
+								ConfigMap: &v1.ConfigMapVolumeSource{
+									DefaultMode: &mode,
+									LocalObjectReference: v1.LocalObjectReference{
+										Name: "bosh-dns",
+									},
+									Items: []v1.KeyToPath{
+										{Key: "Corefile", Path: "Corefile"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, obj := range []v12.Object{&configMap, &deployment, &service} {
+		err := setOwner(obj)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err := controllerutil.CreateOrUpdate(ctx, c, &configMap, configMapMutateFn(&configMap))
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, c, &deployment, deploymentMapMutateFn(&deployment))
+	if err != nil {
+		return err
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, c, &service, mutate.ServiceMutateFn(&service))
+	if err != nil {
+		return err
+	}
+
+	dns.localDNSIP = service.Spec.ClusterIP
+	return err
+}
+
+var re = regexp.MustCompile(`nameserver\s+([\d\.]*)`)
+
+func (dns *boshDomainNameService) rootDNSIP() string {
+	content, err := ioutil.ReadFile("/etc/resolv.conf")
+	if err != nil {
+		content = []byte{}
+	}
+	return GetNameserverFromResolveConfig(content)
+}
+
+// GetNameserverFromResolveConfig read nameserver from resolve.conf
+func GetNameserverFromResolveConfig(content []byte) string {
+	found := re.FindSubmatch(content)
+	if len(found) == 0 {
+		ctxlog.Errorf(context.TODO(), "No nameserver found in resolve.conf using 1.1.1.1")
```

Comment:
I don't think `context.TODO()` is used correctly here. Shouldn't it be `context.Background()`?